starter-core/src/main/java/io/micronaut/starter/feature/database/DataMongo.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.database;
+
+import io.micronaut.core.annotation.Experimental;
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.Dependency;
+import io.micronaut.starter.feature.Category;
+import io.micronaut.starter.feature.Feature;
+import io.micronaut.starter.feature.FeatureContext;
+import jakarta.inject.Singleton;
+
+/**
+ * Add support for Micronaut Data MongoDB. 
+ 
+ * @author graemerocher
+ */
+@Experimental
+@Singleton
+public class DataMongo implements Feature {
+
+    // TODO: Need to support either sync or async
+    private final MongoSync mongoSync;
+
+    public DataMongo(MongoSync mongoSync) {
+        this.mongoSync = mongoSync;
+    }
+
+    @Override
+    public String getName() {
+        return "data-mongodb";
+    }
+
+    @Override
+    public boolean supports(ApplicationType applicationType) {
+        return true;
+    }
+
+    @Override
+    public String getMicronautDocumentation() {
+        return "https://micronaut-projects.github.io/micronaut-data/3.3.x/guide/#mongo";
+    }
+
+    @Override
+    public String getCategory() {
+        return Category.DATABASE;
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        generatorContext.addDependency(
+            Dependency.builder()
+                .annotationProcessor()
+                .groupId("io.micronaut.data")
+                .artifactId("micronaut-data-document-processor")
+                .version("3.3.0-M1")
+        );
+        generatorContext.addDependency(
+            Dependency.builder()
+                .compile()
+                .groupId("io.micronaut.data")
+                .artifactId("micronaut-data-mongodb")
+                .version("3.3.0-M1")
```

Comment:
The version should be on https://github.com/micronaut-projects/micronaut-starter/blob/3.3.x/starter-core/src/main/resources/pom.xml

Reply:
The plan is for this version to move to the BOM once 3.3.0 is final so I am not sure of the need?

Reply:
Just in case we forget to remove the hardcoded version here. At least we could get a dependabot PR, remember that the version is in the BOM and then remove this. If we don't remove the version here we will always generate apps with it, even if micronaut-data-mongodb is added to the bom

Reply:
when the feature is taken out of preview this feature will need to be modified to remove the `isPreview()` method, at that point it will be fairly obvious that the -M1 hard coded versions should be removed.

On the other hand it is more likely to be forgotten if these versions are instead shifted to another non-visible file that won't be updated until a dependabot PR arrives that we would have to pay attention to.

Reply:
:+1: